subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java

```diff
@@ -123,6 +125,45 @@
    */
   void contextRun(Context context, Task<?> parent, Collection<Task<?>> predecessors);
 
+  /**
+   * The method will attempt to schedule calling method to a running plan.
+   *
+   * This method should be called inside a lambda function,
+   *   so that when it was called, there are already plans started running:
+   *
+   * <pre>
+   *     Task{@code <Void>} logging(String str) {
+   *         return Task.blocking("Logging", () -> {
+   *          sendToLoggingServer(str);
+   *         });
+   *     }
+   *
+   *     Task{@code <String>} fetchUrl = fetchUrl(url);
+   *     fetchUrl.andThen((result) -> {
+   *      logging(result).scheduleAndRun();
+   *      AnalyzeResult(result);
+   *     });
+   *
+   * </pre>
+   *
+   * This method attempt to schedule the calling task to current plan's scheduler
+   *    and it can be executed asynchronously
+   *
+   * If the method was called when no plans are running,  the calling cannot be scheduled
+   *   to any plans, and it will throws Exception in this case.
+   *
+   * @throws UnsupportedOperationException
+   */
+  default void scheduleAndRun() throws UnsupportedOperationException {
+    Context ctx = _executionContext.get();
+
+    if (ctx == null) {
+      throw new UnsupportedOperationException();
```

Comment:
This might be a problem. 
The behavior we need is an API that would use current context if it exists, else execute in a new plan.